backend/connect_client.py

```diff
@@ -1,123 +1,87 @@
-
-    Returns:
-        Any: The JSON response from the API request. dict or array
-
-    Raises:
-        requests.exceptions.HTTPError: If an HTTP error occurs.
-        requests.exceptions.RequestException: If a request error occurs.
-    """
-    tier_account_id = os.getenv("TIER_ACCOUNT_ID")
-    api_host = os.getenv("API_HOST")
-    api_key = os.getenv("API_KEY")
-
-    if not all([api_host, tier_account_id, api_key]):
-        raise Exception("Required environment variables are not set")
-
-    url = f"https://{api_host}{endpoint}"
-    headers = {"Authorization": api_key, "Impersonation": tier_account_id}
-
-    try:
-        if method == "POST":
-            response = requests.post(url, headers=headers)
-        else:
-            response = requests.get(url, headers=headers)
-        response.raise_for_status()
-    except requests.exceptions.HTTPError as http_err:
-        logging.error(f"HTTP error occurred: {http_err}")
-        raise
-    except requests.exceptions.RequestException as err:
-        logging.error(f"Error occurred: {err}")
-        raise
-
-    return response.json()
+import re
+from typing import Any, Dict
+
+from connect.client import AsyncConnectClient
+
+
+class Client:
+    def __init__(self, tier_account_id, api_host, api_key):
+        self.connect_client = AsyncConnectClient(
+            api_key=api_key,
+            endpoint=api_host,
+            default_headers={"Impersonation": tier_account_id},
+        )
+
+    async def get_iframe_details(self) -> dict:
+        """
+        Returns the constructed URL for the iFrame, using the first available page and an authorization code.
+
+        Returns:
+            str: iFrame URL
+        """
+        first_available_page = await self._get_first_page()
+        auth_code = await self._get_auth_code(
+            first_available_page["extension"]["login_uri"]
+        )
+
+        return {
+            "url": f"https://"
+            f"{first_available_page['extension']['hostname']}."
+            f"{first_available_page['extension']['domain']}"
+            f"{first_available_page['extension']['login_uri']}?"
+            f"code={auth_code}&"
+            f"redirect_to="
+            f"{first_available_page['url']}",
+            "label": first_available_page["label"],
+            "icon": "https://www.iana.org/_img/2022/iana-logo-header.svg",
+        }
+
+    async def _get_first_page(self) -> Dict[str, Any]:
+        """
+        Fetches the first page from the "/public/v1/devops/pages" endpoint of the API.
+        If the pages array is empty, it logs an error and raises an exception.
+
+        Returns:
+            Dict[str, Any]: A dictionary represents a page.
+
+        Raises:
+            Exception: If the pages array is empty, an exception is raised with a relevant error message.
+        """
+        page = await self.connect_client("devops").collection("pages").all().first()
```

Comment:
I would suggest to filter by `integration_point = 'customer'`. For the demo we will have just 1 customer page plus other pages.

Reply:
done